umap/static/umap/js/modules/data/layer.js

```diff
@@ -115,11 +129,28 @@ export class DataLayer {
     return this._isDeleted
   }
 
+  /**
+   * When receiving a reference version, discard the future uuid
+   * because the layer is now created on the server.
+   */
+  set _reference_version(version){
+    console.debug("set _reference_version", version)
+    if (version !== undefined) {
+      this.__reference_version = version
```

Comment:
Is there any reason we use double underscore here ?

Reply:
There is a need to react to the change of the `_reference_version`, to be able to unset the `_future_uuid` property.

Specifically, this is useful when receiving a (potentially new) reference version from another peer (this is a signal that a save has been triggered). 

That's why I'm using a setter here. the double underscore is to avoid infinite recursions!